src/main/java/com/microsoft/azure/functions/worker/handler/FunctionWarmupHandler.java

```diff
@@ -0,0 +1,82 @@
+
+    public FunctionWarmupHandler() {
+        super(StreamingMessage::getFunctionWarmupRequest,
+                FunctionWarmupResponse::newBuilder,
+                FunctionWarmupResponse.Builder::setResult,
+                StreamingMessage.Builder::setFunctionWarmupResponse);
+    }
+
+    @Override
+    String execute(FunctionWarmupRequest functionWarmupRequest, FunctionWarmupResponse.Builder builder) {
+        try {
+            WorkerLogManager.getSystemLogger().info("azure function java worker warm up start.");
+            this.javaFunctionBroker.setWorkerDirectory(functionWarmupRequest.getWorkerDirectory());
+            warmupFunctionEnvironmentReloadRequestHandler(functionWarmupRequest);
+            UUID functionId = warmupFunctionLoadRequestHandler(functionWarmupRequest);
+            warmupInvocationRequestHandler(functionId);
+            WorkerLogManager.getSystemLogger().info("azure function java worker warm up completed successfully.");
+        } catch (Exception e) {
+            WorkerLogManager.getSystemLogger().severe("warm up process failed with exception: " + e.getMessage());
+            throw new RuntimeException(e);
+        }
+        return "azure function java worker warm up completed";
+    }
+
+    private void warmupFunctionEnvironmentReloadRequestHandler(FunctionWarmupRequest functionWarmupRequest) throws Exception {
+        FunctionEnvironmentReloadRequest.Builder functionEnvironmentReloadRequestBuilder = FunctionEnvironmentReloadRequest.newBuilder();
+        FunctionEnvironmentReloadRequest functionEnvironmentReloadRequest = functionEnvironmentReloadRequestBuilder.putAllEnvironmentVariables(System.getenv()).build();
+        new FunctionEnvironmentReloadRequestHandler(this.javaFunctionBroker).execute(functionEnvironmentReloadRequest, null);
+        WorkerLogManager.getSystemLogger().info("finish warm up FunctionEnvironmentReloadRequestHandler");
+    }
+
+    private UUID warmupFunctionLoadRequestHandler(FunctionWarmupRequest functionWarmupRequest) throws Exception {
+        FunctionLoadRequest.Builder functionLoadRequestBuilder = FunctionLoadRequest.newBuilder();
+        RpcFunctionMetadata.Builder rpcFunctionMetadataBuilder = RpcFunctionMetadata.newBuilder();
+        rpcFunctionMetadataBuilder.setName(WARM_UP_FUNCTION_NAME);
+        rpcFunctionMetadataBuilder.setEntryPoint(WARM_UP_FUNCTION_ENTRY_POINT);
+        rpcFunctionMetadataBuilder.setScriptFile(functionWarmupRequest.getWorkerDirectory() + WARM_UP_FUNCTION_SCRIPT_FILE);
+        Map<String, BindingInfo> map = new HashMap<>();
+        BindingInfo httpTrigger = BindingInfo.newBuilder().setDirection(BindingInfo.Direction.in).setDataType(BindingInfo.DataType.undefined).setType("httpTrigger").build();
+        map.put("req", httpTrigger);
+        BindingInfo http = BindingInfo.newBuilder().setDirection(BindingInfo.Direction.out).setDataType(BindingInfo.DataType.undefined).setType("http").build();
+        map.put("$return", http);

```

Comment:
can move `map` construction to the top of method, then chain and build objects all in one